application/config/routes.rb

```diff
@@ -30,6 +30,17 @@
     end
   end
 
+  # REPO RESOLVER ROUTES
+  post '/view/repo/resolve' => 'repo_resolver#resolve', as: :repo_resolver
+
+  # REPOSITORY SETTINGS ROUTES
+  resources :repository_settings, path: 'repositories/settings', only: [:index, :create] do
+    collection do
+      put :update
+      delete :destroy
+    end
+  end
```

Comment:
why not using the standard Rails Rest routes for put and delete? couldn't the repo domain be the params[:id] ?

Reply:
No. It was causing errors with the routes for the special characters.
It needs to support all servers, including `http://host.docker.internal:8080`